index.js

```diff
@@ -68,23 +68,35 @@ class HtmlIncludeChunksWebpackPlugin {
 		});
 	}
 
-	getLinkTag(fileName) {
+	getFileUrl(fileName, hash) {
+		let url = `${this.publicPath}${fileName}`;
```

Comment:
Could we perhaps use the URL interface?
```javascript
const url = new URL(`${this.publicPath}${fileName}`);

if (hash) {
	url.searchParams.append('hash', hash);
}

return url.href;
```

Reply:
or, alternatively for the hash search param: `url.searchParams.append(hash, '');`

Reply:
I'm not sure we can use URL for this. It looks like the interface expects a full, well-formed URL string instead of just the path.

i.e., this
```javascript
const { URL } = require("url");
let url = new URL("/whatever");
```
throws this error for me
```
TypeError [ERR_INVALID_URL]: Invalid URL: /whatever
    at onParseError (internal/url.js:219:17)
    at parse (internal/url.js:228:3)
    at new URL (internal/url.js:311:5)
    at Object.<anonymous> (/home/danwoz/x.js:3:11)
    at Module._compile (module.js:653:30)
    at Object.Module._extensions..js (module.js:664:10)
    at Module.load (module.js:566:32)
    at tryModuleLoad (module.js:506:12)
    at Function.Module._load (module.js:498:3)
    at Function.Module.runMain (module.js:694:10)
```

Reply:
Silly me.